x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -295,6 +320,81 @@ public void testCSVNoHeaderMode() throws IOException {
         assertEquals("keyword0,0\r\n", actual);
     }
 
+    public void testOutOfRangeComparisons() throws IOException {
+        final int NUM_SINGLE_VALUE_ROWS = 100;
+        bulkLoadTestData(NUM_SINGLE_VALUE_ROWS);
+        bulkLoadTestData(10, NUM_SINGLE_VALUE_ROWS, false, RestEsqlTestCase::createDocumentWithMVs);
+        bulkLoadTestData(5, NUM_SINGLE_VALUE_ROWS + 10, false, RestEsqlTestCase::createDocumentWithNulls);
+
+        List<String> dataTypes = List.of(
+            "alias_integer",
+            "byte",
+            "short",
+            "integer",
+            "long",
+            // TODO: https://github.com/elastic/elasticsearch/issues/102935
+            // "unsigned_long",
+            // TODO: https://github.com/elastic/elasticsearch/issues/100130
+            // "half_float",
+            // "float",
+            "double",
+            "scaled_float"
+        );
+
+        String lessOrLessEqual = randomBoolean() ? " < " : " <= ";
+        String largerOrLargerEqual = randomBoolean() ? " > " : " >= ";
+        String inEqualPlusMinus = randomBoolean() ? " != " : " != -";
+        String equalPlusMinus = randomBoolean() ? " == " : " == -";
+        // TODO: once we do not support infinity and NaN anymore, remove INFINITY/NAN cases.
+        // https://github.com/elastic/elasticsearch/issues/98698#issuecomment-1847423390
+        String humongousPositiveLiteral = randomBoolean() ? HUMONGOUS_DOUBLE : INFINITY;
+        String nanOrNull = randomBoolean() ? NAN : "to_double(null)";
```

Comment:
A shorter and better alternative to the above is to use `randomFrom` which works with 2+ elements and has the same effect.